java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1710,6 +1710,31 @@ public ColumnVector stringLocate(Scalar substring, int start, int end) {
     }
   }
 
+  /**
+   * Returns a new strings column that contains substrings of the strings in the provided column.
+   * Overloading subString to support default value for end index(-1). It signifies end of string.
+   * @param start first character index to begin the substring(inclusive).
+   */
+  public ColumnVector subString(int start) {
+    return subString(start, -1);
+  }
+
+  /**
+   * Returns a new strings column that contains substrings of the strings in the provided column.
+   * 0-based indexing, If the stop position is past end of a string's length, then end of string is
+   * used as stop position for that string.
+   * @param start first character index to begin the substring(inclusive).
+   * @param end   last character index to stop the substring(exclusive)
+   * @return A new java column vector containing the substrings.
+   */
+  public ColumnVector subString(int start, int end) {
```

Comment:
Can we also add in a version that takes starts and stops as columns?  It exists in the C++ API and it is something that we need for substring in spark.  It will not work out of the box because it does not support negative numbers, but with some other code we can make it work.   It could also be used for substring_index in some cases.

Reply:
Sure, I will add that method as well. I wasn't sure if it was needed.